patches/minecraft/net/minecraft/potion/Potion.java.patch

```diff
@@ -88,6 +88,20 @@
 +        ret.add(new net.minecraft.item.ItemStack(net.minecraft.init.Items.field_151117_aB));
 +        return ret;
 +    }
++    
++    /**
++     * An index that, if different than that of a compared potion effect's potion, will override the default sort order.
++     * If the values are the same, the Vanilla logic alone will be responsible for sort order. Overriding this method is 
++     * useful for potions with changing liquid colors since, without this, the potion effect order would shift in the 
++     * inventory GUI when stacked with others due to Vanilla using liquid color as part of PotionEffect's compareTo logic.
```

Comment:
This description is pretty long, I think it can be cut down to more basic information.
You should also mention some more functional information, such as the default value is 0 and whether higher indexes go earlier or later in the sort order.